Namespace-prepended values trigger compilation errors

Parsing namespace-prepended values triggers a compilation error. I guess this is related to acton rather than acton-yang, but am unsure.
I encountered this error when trying to parse the following in sorespo:
```xml
<network-instance xmlns="urn:nokia.com:srlinux:net-inst:network-instance">
    <type xmlns:srl_nokia-network-instance="urn:nokia.com:srlinux:net-inst:network-instance">
      srl_nokia-network-instance:ip-vrf</type>
...
</network-instance>
```

I was not able to reproduce with a simpler file, strangely. Though the following triggers a compilation error. This is the code triggering the error on sorespo.
```acton
import yang
from yang.Identityref import Identityref

actor main(env):
    def identityref_xml_test():
        ir = Identityref.from_xml(
            "srl_nokia-network-instance:ip-vrf",
            [
                (
                    "srl_nokia-network-instance",
                    "urn:nokia.com:srlinux:net-inst:network-instance"
                )
            ]
        )
        print(ir)
    
    identityref_xml_test()
    env.exit(0)
```

Env:
```
ERROR: internal compiler error: compilation of generated Zig code failed, returned error code1
NOTE: this is likely a bug in acton, please report this at:
NOTE: https://github.com/actonlang/acton/issues/new?template=ice.yaml
NOTE: acton 0.29.1.20260811.3.42
zig stdout:

zig stderr:
Acton Project Builder - building /Users/Alex.Huang-Feng/Projects-DT/test-act/foo
Acton Base Builder
Building in /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base
-- filename : json.c
   full_path: /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/json.c
   dir      : /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/.
   file_path: /json.c
-- filename : xml.c
   full_path: /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/xml.c
   dir      : /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/.
   file_path: /xml.c
-- filename : diff.c
   full_path: /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/diff.c
   dir      : /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/.
   file_path: /diff.c
-- filename : md5.c
   full_path: /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/crypto/hash/md5.c
   dir      : /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/crypto/hash
   file_path: /crypto/hash/md5.c
-- filename : term.c
   full_path: /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/term.c
   dir      : /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/.
   file_path: /term.c
-- filename : process.c
   full_path: /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/process.c
   dir      : /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/.
   file_path: /process.c
-- filename : math.c
   full_path: /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/math.c
   dir      : /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/.
   file_path: /math.c
-- filename : wyhash.c
   full_path: /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/hash/wyhash.c
   dir      : /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/hash
   file_path: /hash/wyhash.c
-- filename : random.c
   full_path: /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/random.c
   dir      : /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/.
   file_path: /random.c
-- filename : fs.c
   full_path: /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/fs.c
   dir      : /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/.
   file_path: /fs.c
-- filename : qcheck.c
   full_path: /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/qcheck.c
   dir      : /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/.
   file_path: /qcheck.c
-- filename : __builtin__.c
   full_path: /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/__builtin__.c
   dir      : /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/.
   file_path: /__builtin__.c
-- filename : time.c
   full_path: /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/time.c
   dir      : /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/.
   file_path: /time.c
-- filename : testing.c
   full_path: /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/testing.c
   dir      : /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/.
   file_path: /testing.c
-- filename : argparse.c
   full_path: /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/argparse.c
   dir      : /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/.
   file_path: /argparse.c
-- filename : base64.c
   full_path: /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/base64.c
   dir      : /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/.
   file_path: /base64.c
-- filename : net.c
   full_path: /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/net.c
   dir      : /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/.
   file_path: /net.c
-- filename : http.c
   full_path: /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/http.c
   dir      : /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/.
   file_path: /http.c
-- filename : file.c
   full_path: /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/file.c
   dir      : /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/.
   file_path: /file.c
-- filename : re.c
   full_path: /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/re.c
   dir      : /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/.
   file_path: /re.c
-- filename : buildy.c
   full_path: /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/buildy.c
   dir      : /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/.
   file_path: /buildy.c
-- filename : uri.c
   full_path: /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/uri.c
   dir      : /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/.
   file_path: /uri.c
-- filename : rts.c
   full_path: /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/acton/rts.c
   dir      : /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/acton
   file_path: /acton/rts.c
-- filename : snappy.c
   full_path: /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/snappy.c
   dir      : /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/.
   file_path: /snappy.c
-- filename : logging.c
   full_path: /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/logging.c
   dir      : /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/base/out/types/.
   file_path: /logging.c
Debug build
Threads enabled
Acton Standard Library Builder
Building in /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.local/lib/acton/std
Debug build
Threads enabled
error: invalid option: -Dacton_root_stubs
error: invalid option: -Dacton_modules
/Users/Alex.Huang-Feng/.local/lib/acton/zig/lib/std/Build.zig:2066:35: 0x100913777 in dependency__anon_132407 (build)
            return dependencyInner(b, name, pkg.build_root, if (@hasDecl(pkg, "build_zig")) pkg.build_zig else null, pkg_hash, pkg.deps, args);
                                  ^
/Users/Alex.Huang-Feng/Projects-DT/test-act/foo/build.zig:65:36: 0x1007846bb in build (build)
    const actdep_std = b.dependency("std", .{
                                   ^
/Users/Alex.Huang-Feng/.local/lib/acton/zig/lib/std/Build.zig:2264:33: 0x1007840f3 in runBuild__anon_28527 (build)
        .void => build_zig.build(b),
                                ^
/Users/Alex.Huang-Feng/.local/lib/acton/zig/lib/compiler/build_runner.zig:463:29: 0x1005eff3b in main (build)
        try builder.runBuild(root);
                            ^
/Users/Alex.Huang-Feng/.local/lib/acton/zig/lib/std/start.zig:699:88: 0x1005f21d7 in callMain (build)
    if (fn_info.params[0].type.? == std.process.Init.Minimal) return wrapMain(root.main(.{
                                                                                       ^
???:?:?: 0x1813ebdff in start (/usr/lib/dyld)
Acton Project Builder - building /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/../../../../../Users/Alex.Huang-Feng/.cache/acton/deps/yang-N-V-__8AAI4cJwByJO3YK_rs17D_K2hXVKcK7UBL2ikUiuUy
Debug build
Threads enabled
Debug build
Threads enabled
Building executable from: /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/out/types/foo.root.c -> foo
install
+- install ActonProject
   +- compile lib ActonProject Debug aarch64-macos-none 3 errors
/Users/Alex.Huang-Feng/Projects-DT/test-act/foo/./out/types/foo/foo.h:6:10: error: non-portable path to file '"out/types/yang/identityref.h"'; specified path differs in case from file name on disk
#include "out/types/yang/Identityref.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/Alex.Huang-Feng/Projects-DT/test-act/foo/out/types/foo/foo.c:3:10: note: in file included from /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/out/types/foo/foo.c:3:
#include "out/types/foo/foo.h"
         ^
src/foo.act:7:96: error: use of undeclared identifier 'yangQ_IdentityrefQ_IdentityrefG_methods'

                                                                                               ^
src/foo.act:59:5: error: call to undeclared function 'yangQ_IdentityrefQ___init__'; ISO C99 and later do not support implicit function declarations

    ^
error: 3 compilation errors
failed command: /Users/Alex.Huang-Feng/.local/lib/acton/zig/zig build-lib -cflags -ffile-prefix-map=/Users/Alex.Huang-Feng/Projects-DT/test-act/= -DDEV -DACTON_THREADS -fwrapv -- /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/out/types/foo/foo.c -ODebug -target aarch64-macos-none -mcpu apple_m1 -I /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/. -I /Users/Alex.Huang-Feng/.cache/acton/zig-local-cache/o/d2a8807310aad5e4684741c6889382d5 -I /Users/Alex.Huang-Feng/.cache/acton/zig-local-cache/o/bd46430b7bacc53fe9beb2aa53804cd2 -I /Users/Alex.Huang-Feng/.cache/acton/zig-local-cache/o/3f93f0e41420b425db3389d582d2f8d1 -Mroot -lc++ -lc --cache-dir /Users/Alex.Huang-Feng/.cache/acton/zig-local-cache --global-cache-dir /Users/Alex.Huang-Feng/.cache/acton/zig-global-cache --name ActonProject -static --zig-lib-dir /Users/Alex.Huang-Feng/.local/lib/acton/zig/lib/ --listen=-

Build Summary: 40/45 steps succeeded (1 failed)
install transitive failure
+- install ActonProject transitive failure
|  +- compile lib ActonProject Debug aarch64-macos-none 3 errors
+- install foo transitive failure
   +- compile exe foo Debug aarch64-macos-none transitive failure
      +- compile lib ActonProject Debug aarch64-macos-none (+9 more reused dependencies)
      +- compile lib ActonProject Debug aarch64-macos-none (+9 more reused dependencies)

error: the following build command failed with exit code 1:
/Users/Alex.Huang-Feng/.cache/acton/zig-local-cache/o/c52b41696b3eeabee65d940bde691eef/build /Users/Alex.Huang-Feng/.local/lib/acton/zig/zig /Users/Alex.Huang-Feng/.local/lib/acton/zig/lib /Users/Alex.Huang-Feng/Projects-DT/test-act/foo /Users/Alex.Huang-Feng/.cache/acton/zig-local-cache /Users/Alex.Huang-Feng/.cache/acton/zig-global-cache --seed 0xd6d34857 -Zc30b5f6683475e66 --prefix /Users/Alex.Huang-Feng/Projects-DT/test-act/foo/out --prefix-exe-dir bin -Dtarget=aarch64-macos-none -Dcpu=apple_m1 -Doptimize=Debug -Dacton_modules=out/types/foo/foo.c -Dacton_root_stubs=out/types/foo.root.c -Dacton_libraries=
```

--- a/src/identityref.act
+++ b/src/identityref.act
@@ -41,7 +41,7 @@ class Identityref:
 
     @staticmethod
     def from_xml(val: str, nsdefs: list[(?str, str)]):
-        parts = val.split(":")
+        parts = val.strip().split(":")
         if len(parts) == 1:
             return PartialIdentityref(parts[0])
         elif len(parts) == 2:

--- a/src/test_data.act
+++ b/src/test_data.act
@@ -1510,6 +1510,13 @@ def _test_identityref():
     except YangValidationError as e:
         testing.assertEqual(str(e), "Invalid value at /le3: 'id8' - expected identityref with bases: ['aa:id8']")
 
+    # Spaces in values
+    xml_in = xml.decode("""<data><le2 xmlns="urn:cesent:mod1" xmlns:aa="urn:cesent:mod1">         aa:id8          </le2></data>""")
+    gd = from_xml(s, xml_in)
+    exp = ygdata.Container({
+        ygdata.Id("urn:cesent:mod1", "le2"): ygdata.Leaf(Identityref('id8', ns='urn:cesent:mod1', mod='mod1', pfx='aa'), ns='urn:cesent:mod1', module='mod1')
+    })
+    testing.assertEqual(exp.prsrc(), gd.prsrc())
 
 def _test_xml_path_basic():
     """Test basic XML path navigation, mirrors _test_json_path_basic"""

